atc/lidar/scanner.go

```diff
@@ -58,13 +69,30 @@ func (s *scanner) Run(ctx context.Context) error {
 			}()
 			defer waitGroup.Done()
 
-			s.check(spanCtx, resource, resourceTypes, resourceTypesChecked)
+			s.check(ctx, resource, resourceTypes, resourceTypesChecked)
 		}(resource, resourceTypes)
 	}
+}
 
-	waitGroup.Wait()
+func (s *scanner) scanUncheckedResourceTypes(ctx context.Context, resourceTypes db.ResourceTypes, waitGroup *sync.WaitGroup, resourceTypesChecked *sync.Map) {
+	logger := lagerctx.FromContext(ctx)
+	for _, resourceType := range resourceTypes {
+		waitGroup.Add(1)
 
-	return nil
+		go func(resourceType db.ResourceType, resourceTypes db.ResourceTypes) {
+			defer func() {
+				err := util.DumpPanic(recover(), "scanning resource type %d", resourceType.ID())
```

Comment:
A bit confused about when this could panic. I guess it's just being defensive so as not to take down the ATC due to a programming mistake - but feels like that should get caught in tests, as it's just relatively simple control flow to create a build in the DB.

p.s. I recognize this panic recovery was present in the scanner beforehand - happy to keep it, just feels a bit odd to me

Reply:
I say we keep it. Came from this PR: https://github.com/concourse/concourse/pull/5842